File: Examples/CSharp/Articles/ExtractImagesFromWorksheets.cs
using System.IO;
using System;
using Aspose.Cells;
using Aspose.Cells.Rendering;

namespace Aspose.Cells.Examples.CSharp.Articles
{
    public class ExtractImagesFromWorksheets
    {
        public static void Run()
        {
            //Source directory
            string sourceDir = RunExamples.Get_SourceDirectory();

            //Output directory
            string outputDir = RunExamples.Get_OutputDirectory();

            // Open a template Excel file
            Workbook workbook = new Workbook(sourceDir + "sampleExtractImagesFromWorksheets.xlsx");

            // Get the first worksheet
            Worksheet worksheet = workbook.Worksheets[0];

            // Get the first Picture in the first worksheet
            Aspose.Cells.Drawing.Picture pic = worksheet.Pictures[0];

            // Set the output image file path
            string picformat = pic.ImageType.ToString();
            
            // Note: you may evaluate the image format before specifying the image path
            // Define ImageOrPrintOptions
            ImageOrPrintOptions printoption = new ImageOrPrintOptions();

            // Specify the image format
            printoption.ImageType = Drawing.ImageType.Jpeg;
            
            // Save the image
            pic.ToImage(outputDir + "outputExtractImagesFromWorksheets.jpg", printoption);

            Console.WriteLine("ExtractImagesFromWorksheets executed successfully.\r\n");
        }
    }
}

File: Examples/VisualBasic/Articles/ExtractImagesFromWorksheets.vb
Imports Microsoft.VisualBasic
Imports System.IO

Imports Aspose.Cells
Imports Aspose.Cells.Rendering

Namespace Aspose.Cells.Examples.Articles
    Public Class ExtractImagesFromWorksheets
        Public Shared Sub Main()
            ' The path to the documents directory.
            Dim dataDir As String = Aspose.Cells.Examples.Utils.GetDataDir(System.Reflection.MethodBase.GetCurrentMethod().DeclaringType)

            'Open a template Excel file
            Dim workbook As New Workbook(dataDir & "book1.xls")
            'Get the first worksheet
            Dim worksheet As Worksheet = workbook.Worksheets(0)
            'Get the first Picture in the first worksheet
            Dim pic As Global.Aspose.Cells.Drawing.Picture = worksheet.Pictures(0)
            'Set the output image file path
            Dim fileName As String = dataDir & "aspose-logo.Jpg"
            Dim picformat As String = pic.ImageFormat.ToString()
            'Note: you may evaluate the image format before specifying the image path

            'Define ImageOrPrintOptions
            Dim printoption As New ImageOrPrintOptions()
            'Specify the image format
            printoption.ImageFormat = System.Drawing.Imaging.ImageFormat.Jpeg
            'Save the image
            pic.ToImage(fileName, printoption)


        End Sub
    End Class
End Namespace